Marlin/src/HAL/STM32/timers.cpp

```diff
@@ -31,16 +31,20 @@
 
 #define NUM_HARDWARE_TIMERS 2
 
-#ifndef SWSERIAL_TIMER_IRQ_PRIO
-  #define SWSERIAL_TIMER_IRQ_PRIO 1
-#endif
 #ifndef STEP_TIMER_IRQ_PRIO
   #define STEP_TIMER_IRQ_PRIO 2
 #endif
 #ifndef TEMP_TIMER_IRQ_PRIO
   #define TEMP_TIMER_IRQ_PRIO 14   // 14 = after hardware ISRs
 #endif
 
+#if HAS_TMC_SW_SERIAL
+  #include <SoftwareSerial.h>
```

Comment:
Do we sometimes want to include this from the platform packages? If not, then we can use quotes here to use the local (quoted relative path) include.

Reply:
Eventually our copy of SoftwareSerial will go away, and we will use the version from the STM32 Arduino framework. We need to upgrade that dependency first, but I think other issues (not sure what they are) are preventing that from happening. This priority will still need to be set once we are using the framework version.

So...quotes would work for now, but eventually this will need to include from an external library.

Reply:
It does appear that some of the STM32 environments are including the package-provided SoftwareSerial, but I haven't checked whether any of those environments then need to use it.

Reply:
Ah, I see this is a standard idiom, so the pointy braces are expected.

Reply:
> Eventually our copy of SoftwareSerial will go away

It looks like we wanted to have total control over which timers are used. Maybe that's not as important now, and we have all potential conflicts sorted out…?

Reply:
I just realized that our current dependency versions are allowing versions of the STM32 Arduino Framework which may allow us to delete our own copy of SoftwareSerial.cpp. I was thinking we were only getting versions which didn't even have this priority interface yet.

I suggest just leaving the angle brackets for now to un-break everybody, and I will investigate whether we can go back to the library version now that it has all of our fixes in it.

Reply:
Nevermind, the versioning on those framework components is confusing, and the version needed is the one that broke CI a while back. I'll poke at it a little to see if there is an easy solution for that, but am not sure whether I'll manage to solve it.

Reply:
> It looks like we wanted to have total control over which timers are used. Maybe that's not as important now, and we have all potential conflicts sorted out…?

Timers weren't actually the reason we pulled this in. The framework version was very much broken. It might have worked for 2-wire, but was broken for 1-wire. That made it much worse than using FYSETC/SoftwareSerialM.

@LinoBarreca and I worked to fix SoftwareSerial in the framework, and put a copy here as a stop-gap until we could pull in the new framework.

Changing the timer number for this HAL has to happen in the variant.h file, because that is baked into the code in the framework.

I actually have it all building with the updated framework. I'll post a PR and see if I can round up some people to try it out.